### ISSUE TITLE
version banners

### ISSUE THREAD

**Issue by shawna-slh** [w3c/wai-wcag-em-report-tool#48]
Here is a proposal. Feel free to implement as is, or suggest changes.

Style:
* yellow banner at the very top, e.g., https://wai-website-theme.netlify.app/components/colors/
* text weight normal (not bold)

## Now

Text for existing/old tool:
> A redesign of this WCAG-EM Report Tool will be published here in August or September 2021. You can see the [Redesign Preview](https://wcag-em-report-tool-2021-redesign.netlify.app//) and [changelog of revisions](https://github.com/w3c/wai-wcag-em-report-tool/wiki/What's-new-in-the-2021-redesign).

## When publish

Old tool banner text:
> An updated redesign of this WCAG-EM Report Tool is available at [https://www.w3.org/WAI/eval/report-tool/](https://www.w3.org/WAI/eval/report-tool/#) ([changelog](https://github.com/w3c/wai-wcag-em-report-tool/wiki/What's-new-in-the-2021-redesign) of revisions).

New tool banner text:
> The previous version of this WCAG-EM Report Tool is available at [@@](@@) ([changelog](https://github.com/w3c/wai-wcag-em-report-tool/wiki/What's-new-in-the-2021-redesign) of revisions).

**Comment by hidde** [comment]
Looks good to me, I will go and add these ASAP! (@nitedog if you're good too?)

Example of the style: <img width="1657" alt="yellow banner with non bolded black text: A redesign of this WCAG-EM Report Tool will be published here in August or September 2021. You can see the Redesign Preview and changelog of revisions." src="https://user-images.githubusercontent.com/178782/127516538-7dce0f90-76f3-4f6f-91e2-77b0c7cf29ae.png">

**Comment by shawna-slh** [comment]
Cool. (in earlier 1:1, @nitedog says OK)

That text is hard to read, but maybe just because it's smaller than in real life? I'm still inclined to go with not bold.

I'm just now noticing that the link color on yellow is hard to read. It's 4.6:1 so OK for AA but not AAA :-(

If we used the other blue --  #003366 -- for unvisited links on that yellow background, then it would be 8.2:1.

Are you OK doing that, Hidde?

**Comment by nitedog** [comment]
I'm OK with implementing ASAP, and with Shawn's suggestions if Hidde accepts them.

**Comment by hidde** [comment]
The banner is now on https://www.w3.org/WAI/eval/report-tool/

**Comment by hidde** [comment]
The banner is now also on the [3.0.0 version](https://wcag-em-report-tool-2021-redesign.netlify.app/).

@shawna-slh I feel this banner grabs too much attention, and thinking about alternatives, could we maybe do something like this: 

<img width="658" alt="Status section of the footer, with links to Changelog and link to Previous version, just after the link to Changelog" src="https://user-images.githubusercontent.com/178782/128015084-e4d8d5bf-12b1-4c54-9151-650447697dc8.png">

**Comment by shawna-slh** [comment]
I agree the top banner is too much to be there all of the time.

However, I think we need it much more clear than the little text in the footer.
(+1 for adding it there for redundancy)

**_User Experience:_** Jan is doing their company's annual evaluation of websites using the WCAG-EM Report Tool. Jan goes to the URI Jan has bookmarked. It looks totally different from what Jan is used to. Jan thinks something is wrong. (maybe Jan gets confused, frustrated, etc...)

What about if we had it prominently with a dismiss option? Examples on the WAI website:
* https://www.w3.org/WAI/ Making the Web Accessible "Hide Section -"
* https://www.w3.org/WAI/fundamentals/accessibility-intro/ "Hide Options -"

Except when you dismiss it, it's gone for good, not just a toggle.

In that case, we could consider having it other than the banner at the top &mdash; e.g., maybe before of after the h1 Overview, rather than at the top.

<hr>

Another brainstorm is to have it only on the Overview page.

I'm happy to work through options as useful.

**Comment by hidde** [comment]
what about an inline warning with dismiss button ('Hide')?

<img width="1138" alt="box under h1 in warning style which says previous tool is available at url, after that a button with the text ‘Hide’" src="https://user-images.githubusercontent.com/178782/128368823-61384913-fa43-492e-a395-4718261a90f9.png">

**Comment by shawna-slh** [comment]
I really like this! Do you?

Since we use "Hide" as a toggle other places and here it will dismiss it forever, what do you think about a different label for the button? maybe
* "Dismiss" (maybe too jargony)
*  "Close"

Or, is not having the [-] after it enough of a clue?

**Comment by hidde** [comment]
For context, I was planning for this behavior: when it is clicked, we set a cookie or localstorage, so that users don't see it again in that browswer/device. 

I guess you're right that Hide already does something else in other places… I like 'Dismiss'… 'Don't show again' could also work? 

'Hide forever' 😅?

**Comment by shawna-slh** [comment]
> For context, I was planning for this behavior: when it is clicked, we set a cookie or localstorage, so that users don't see it again in that browswer/device.

+1

That was the way the other Hides were suppose to work. It looks like the Hide Options for the translation list is working, but the Hide Section on the home page is not. :(

> I like 'Dismiss'… 'Don't show again' could also work?

I'm fine with either.

Remember to also add the link in the footer.

**Comment by hidde** [comment]
Have implemented: https://wcag-em-report-tool-2021-redesign.netlify.app/ 

Now… and I'm sorry… now that I have it implemented, I don't really like it. It distracts more than I hoped and seems to communicate “there's something wrong / alarming”, where, really, there's something good, we've improved the old version. So… my vote would be to only have 'Previous version' in the footer, because I assume there are a lot more people who don't need to find this link, than people who do. What do you think, @shawna-slh?

cc @nitedog for visibility and maybe input?

**Comment by shawna-slh** [comment]
> now that I have it implemented, I don't really like it.

I fully understand that seeing something in context can change ones perspective. (Keep that in mind for others' reviews of other projects :-)

I agree that it doesn't really communicate what we want. I do think we need something there, though.

Some people will be thrown off by the significant UI change. See [Jan's user experience above](https://github.com/w3c/wai-wcag-em-report-tool/issues/48#issuecomment-892173782). I am one of those people who would be initially freaked out and then really annoyed. Maybe Jan and I make up only 0.02% of users? maybe 20%? maybe 80%? We don't know and not worth trying to gather that data.

Since it is now dismissable, what do y'all think of using more words to better communicate what we want to get across? maybe something along the lines of:
> We updated this tool in September 2021. You can see the list of revisions in the [changelog](@@). If you really want to use the older version of the tool, it’s available at [https://w3c.github.io/wcag-em-report-tool/](https://w3c.github.io/wcag-em-report-tool/).

And maybe even more info about that it works with reports from the previous version.

?

**Comment by hidde** [comment]
I would prefer less words to more words. In my view, more words means:

* less likely that people will read them
* note takes more space, literally and figuratively

As said above, I'd say we only need “Previous version” link in the footer. In addition, to address Jan's concern, and yours, why don't we add this information in our announcements? Maybe the announcement can contain some sort of FAQ that answers these questions?

If we are confident 3.0.0 is an improvement over 2.0.0, do we really need the note? If we're not confident the new thing is better, we should probably not launch until we are?

**Comment by shawna-slh** [comment]
> to address Jan's concern, and yours, why don't we add this information in our announcements?

We will include it in the announcements; however, **few people will see them**. We cannot count on tool users seeing the announcements.

I'm missing why we would not want clear information in a banner to address this common user experience, and try to **avoid freaking out our users**?

_(Also, from a WAI site design perspective, I would like us to use the visual design that we previously agreed on with a small yellow border. That is more appropriate with how we are presenting information throughout the WAI website.)_

From [27 August EOWG minutes](https://www.w3.org/2021/08/27-eo-minutes#t01):
> * Somebody may want to know where is the old version especially if they have a project in progress.
> * It is important that people submitting with the old version can finish the submission.
> * If I save a json file with the old tool. I can import it with the new version and finish the submission?
> * I think that it is important to let people know that this is a new version and indicate a deadline.
> * Yes, it is important to indicate that the old version will move away but people will have time to still use it.

Taking all EOWG input into consideration, here's a straw proposal:

[[
Welcome to the redesigned WCAG-EM Report Tool. ([Changelog](https://github.com/w3c/wai-wcag-em-report-tool/wiki/Changelog:-What's-new-in-the-2021-redesign-of-the-WCAG-EM-Report-Tool)) **You can import data from the previous tool into this redesign.** We plan to leave the [old design](https://w3c.github.io/wcag-em-report-tool/) available through November 2021. [Dismiss]
]]

**Comment by nitedog** [comment]
I think the use of **import data** is confusing here -- this means adding data from automated tools in Step 4 of the tool. The better wording would be:

[[
Welcome to the redesigned WCAG-EM Report Tool. (Changelog) You can open reports from the previous tool into this redesign. We plan to leave the old design available through November 2021. [Dismiss]
]]

PS: I thought the old version will remain available indefinitely on GitHub, just the banner will be removed after ~3 months?

**Comment by hidde** [comment]
Shawn said:

>>  to address Jan's concern, and yours, why don't we add this information in our announcements?
>
> We will include it in the announcements; however, few people will see them. We cannot count on tool users seeing the announcements.

Which is why we still have the banner.

> I'm missing why we would not want clear information in a banner to address this common user experience, and try to avoid freaking out our users?
> 
> (Also, from a WAI site design perspective, I would like us to use the visual design that we previously agreed on with a small yellow border. That is more appropriate with how we are presenting information throughout the WAI website.)

My assumption, and what I gathered from EOWGs response, is that our main priority here is not “not freaking out users“, as strange as that may sound. 

By having a banner that looks like a warning, we would signal that this tool is not ready or we're not confident of the qualities of the new tool. EOWG didn't feel that level of warning was warranted and echoed it should mostly be about being clear about discontinuation of the old tool. This is why we came up with the new banner during the meeting: it looks _not_ like a warning, yet it is clear about expectations. 

>  Welcome to the redesigned WCAG-EM Report Tool. (Changelog) You can import data from the previous tool into this redesign. We plan to leave the old design available through November 2021. [Dismiss]

Thanks for providing an all new description, but the current one was approved by EOWG. Which parts of the current description are not good and would you like to change? Note that if we want to change this, we would need to take it to EOWG again, this may not be worthwile and could push release back more? 

I would suggest we stick with the current, unless any specific part of that causes significant problems. In more detail: “redesigned” is longer than “new”, covers the same; info about import would probably be more relevant if import would _cease_ to work with this new tool, so would leave it out; “we plan to” makes the text less definite, while EOWG expressed making it definite is important to them.

Shadi said:

> PS: I thought the old version will remain available indefinitely on GitHub, just the banner will be removed after ~3 months?

In EOWG we concluded not needed, we could reopen that discussion though.

**Comment by sharronrush** [comment]
I affirm that the group was unanimous in their rejection of the yellow banner and the feeling that ti communicated a lack of confidence in the new tool. We can take additional comments about it but the first intention was to completely get rid of it and so this was, we agreed a good compromise. It tells folks of the change and sets expectation for when it will no longer be available. Let's try not to make to much of this - it was felt by all to be completely adequate information and much less alarming that the yellow caution banner.

**Comment by shawna-slh** [comment]
Hi all. Here is some more context that might be helpful.

For information on other versions, the WAI website design specifies a box with a small yellow border. For example:
* **Links to other languages (used on many pages):** https://www.w3.org/WAI/fundamentals/accessibility-intro/#languagehead
* Detailed info on status and link to newer info: https://www.w3.org/WAI/standards-guidelines/wcag-mobile-overlap/
* Simple archive message with link:  https://www.w3.org/WAI/business-case/archive/soc

If there is a strong reason to deviate from that established style, it would be good to use another existing style. For example, here we use a more-info-type box for the "Dated" notice: https://www.w3.org/WAI/policies/#main

**Comment by shawna-slh** [comment]
> yellow banner and the feeling that ti communicated a lack of confidence in the new tool

Yes, I *totally* agree with @Hidde and the folks in last week's EOWG meeting that "The previous version of this WCAG-EM Report Tool is available at ... (changelog of revisions)." in yellow outline is not good!

I think the change to "Welcome to..." takes care of that.

**Comment by shawna-slh** [comment]
I was trying to incorporate EOWG input. After more input from Hidde, Shadi, and Sharron, here are some more thoughts:
* changelog - Maybe not important? How many people would want to see a list of changes?
* date - I don't see why we need it here? It will be on the old version. And current plans are to leave the old version up indefinitely.

Based on that, here are updated ideas:

1. shorter
> Welcome to the redesigned tool. You can open old reports with this redesign. The previous design is still available. [Dismiss]

2. more chatty
> Welcome to the redesigned WCAG-EM Report Tool. You can use this redesigned tool for new reports and to open reports started in the old design. (For those who don't like change, you can still use the previous design.) [Dismiss]

**Comment by hidde** [comment]
1) and 2) reintroduce the problem EOWG and I wanted to address and undoes their solutions: 

* the message was not definitive; solution: we added the end date. EOWG was agreeing an end date would be good, 3 months was a straw proposal that everyone seemed to like)
* it felt like we were saying people might want to switch to or prefer the old version, this is not what we want to say, as this new tool is meant to be an improvement that people like more. Solution: we removed 'if you really want to use the old version'

**Comment by nitedog** [comment]
@hidde thank you for clarifying the EOWG discussion, which I had also missed!

I'm not stuck on specific wording but feel quite strongly that it is important to communicate "you can open your existing reports with this tool" in the notification. I see that this has been raised by others in the [EOWG minutes of 27 August](https://www.w3.org/2021/08/27-eo-minutes) but don't feel it has been sufficiently implemented in the final wording:

> **Krisanne:** It is important that people submitting with the old version can finish the submission.
> **Hidde:** Someone that is submitting can export on json file and finish the submission.
> <Donna> The export/import of reports: is that well documented?
> **Sharron:** If I save a json file with the old tool. I can import it with the new version and finish the submission?
> **Hidde:** yes it should work, we have tested and debugged problems we found.
> **Sharron:** Maybe this can be announced to the audience?
> **Hidde:** yes

In my opinion, this point is not announced/communicated to the audience as agreed. Maybe the minutes are incomplete or I missed something, please feel free to clarify.

Note, also, that the currently proposed notice uses "WCAG-EM Tool", which is not the correct name of the tool and could therefore be confusing to some readers.

Putting that all together, how about this wording:

> Welcome to the new WCAG-EM Report Tool. You can open your existing reports with this tool. The old version is available until November 27, 2021. [Dismiss]

PS: I don't really think the date is necessary, especially given that it is so far out, but see that this seems to be important to others in the group.

**Comment by shawna-slh** [comment]
> Welcome to the new WCAG-EM Report Tool.

I'm uncomfortable with "new". I think most people who read "new" will take that to mean that it is a **new tool**. (The tool has been around for many years. Of course, regular users will know it's been around for a while, but others will not.)

Shadi pointed out that it's more than a "redesign" -- some new features and all new code.

Does "updated" convey what users will want to know, and we want them to know?

I also know that some people will want to know what was updated. I don't know how many?

Maybe a smooth way to provide that is to link "updated" to the changelog?

**Comment by sharronrush** [comment]
These are terrific suggestions @shawna-slh 
+1 to linked "updated" reference

**Comment by hidde** [comment]
Hidde, Shadi and Shawn met last week and got to an updated wording. Unfortunately I only saved it in Zoom chat, I have tried to rephrase it: 

> Welcome to the updated WCAG-EM Report Tool. The old version is available until 27 November 2021. You can also open existing reports into this tool.

**Comment by shawna-slh** [comment]
Hi @Hidde I think you and I were going to followup in a short meeting? Here's what I have from the end of the meeting last week:

**Welcome to the [updated](changelog) WCAG-EM Report Tool. You can open your existing reports with this tool. The [old version](old) is available, and is no longer supported. [Dismiss]**

(or: Welcome to the [updated](changelog) WCAG-EM Report Tool. You can open your existing reports with this tool. The [old version](old) is no longer supported. [Dismiss])

Instead of further comment in this issue, can we talk this morning?

**Comment by hidde** [comment]
Thanks, either one works fine for me, I do prefer the second. Not having the date makes it less definite, so will have to run by EOWG I think.

**Comment by hidde** [comment]
Discussed in EOWG 8 September, we'll use: 

> Welcome to the updated WCAG-EM Report Tool. You can open your existing reports with this tool. The previous version is no longer supported. [Dismiss]

(replaced 'old' with 'previous')

**Comment by hidde** [comment]
@shawna-slh @nitedog The updated banner text is [live](https://wcag-em-report-tool-2021-redesign.netlify.app/), please let me know if we can close this issue.

**Comment by nitedog** [comment]
many thanks, works well for me!

**Comment by hidde** [comment]
Thanks @nitedog ! 

@shawna-slh Please let me know if you're happy with it too and I'll close the issue, thanks!

**Comment by shawna-slh** [comment]
Thanks for making the changes I sent separately. However, the design issue is still open.
